src/Cake.NuGet/NuGetPackageInstaller.cs

```diff
@@ -175,6 +160,38 @@ public IReadOnlyCollection<IFile> Install(PackageReference package, PackageType
             return result;
         }
 
+        private static DirectoryPath GetPackagePath(DirectoryPath root, PackageReference package)
+        {
+            if (package.Parameters.ContainsKey("version"))
+            {
+                var version = package.Parameters["version"].First();
+                return root.Combine($"{package.Package}.{version}");
```

Comment:
Well rule is that nuget displays x.x.x but if version element has x.x.x.x it'll be used for folder name :)